### ISSUE TITLE
Incorrect types for sqlalchemy join and distinct Query methods

### ISSUE THREAD

**Issue by MarcDufresne** [python/typeshed#7624]
Hi all, PyCharm [has integrated Typeshed starting from version 2022.1](https://www.jetbrains.com/help/pycharm/type-hinting-in-product.html#typeshed) and we've noticed that it now highlights usages of SQLA's `Query.join` and `Query.distinct` (and potentially others) as returning `None`. This is incorrect when comparing with the actual source code and documentation of SQLAlchemy.

Here is the Typeshed definition:

```python
# typeshed/stubs/SQLAlchemy/sqlalchemy/orm/query.pyi

class Query(...):
    ...
    def join(self, target, *props, **kwargs) -> None: ...
    def distinct(self, *expr) -> None: ...
```

And here are the links to the SQLA documentation:
- [`Query.join`](https://docs.sqlalchemy.org/en/14/orm/query.html?highlight=join#sqlalchemy.orm.Query.join)
- [`Query.distinct`](https://docs.sqlalchemy.org/en/14/orm/query.html?highlight=join#sqlalchemy.orm.Query.distinct)

Both mention that these methods return a new `Query` object

Here's a screenshot of a usage of `distinct`:
![image](https://user-images.githubusercontent.com/599102/163429039-d4946674-35ac-49ee-b8ed-0a3b3ac99933.png)

Let me know if more details are needed, thanks!

**Comment by srittau** [comment]
It seems the PyCharm stubs are fairly outdated. This was fixed in #7237 and uploaded as types-SQLAlchemy 1.4.15 to PyPI.

**Comment by MarcDufresne** [comment]
Alright thanks for the quick confirmation, I'll open a ticket on the PyCharm issue tracker
